### PR TITLE
Save and revert only if there areremapped datablocks from make_paths_absolute

### PR DESCRIPTION
**Pull request by kaamaurice**
## Changelog Description
- `make_paths_absolute` from `openpype.hosts.blender.api.utils` return remapped datablocks set
- `make_paths_absolute` script save and revert only if there are remapped datablocks

### CODE REVIEW

File: openpype/hosts/blender/utility_scripts/make_paths_absolute.py [hunk 1]
```diff
@@ -25,11 +25,11 @@
         nargs="?",
         help="source filepath",
     )
-    args = parser.parse_args(sys.argv[sys.argv.index("--") + 1 :])
+    args = parser.parse_args(sys.argv[sys.argv.index("--") + 1:])
```
Comment (Tilix4) [inline openpype/hosts/blender/utility_scripts/make_paths_absolute.py]: I think we better do:
```suggestion
    args = parser.parse_known_args(sys.argv[sys.argv.index("--") + 1:])
```

File: openpype/hosts/blender/api/utils.py [hunk 2]
```diff
@@ -343,27 +343,47 @@ def make_paths_absolute(source_filepath: Path = None):
...
-        try:
-            if datablock and datablock.filepath.startswith("//"):
+    relative_datablocks = set()
+    for data_name in dir(bpy.data):
+        data_collection = getattr(bpy.data, data_name)
+        if not isinstance(data_collection, bpy.types.bpy_prop_collection):
+            continue
+        for datablock in data_collection.values():
+            if not hasattr(datablock, "filepath"):
+                break
```
Comment (Tilix4) [inline openpype/hosts/blender/api/utils.py]: Don't you mean `continue` ?

File: openpype/hosts/blender/api/utils.py [hunk 3]
```diff
@@ -343,27 +343,47 @@ def make_paths_absolute(source_filepath: Path = None):
...
                             start=source_filepath.parent,
                         )
                     ).resolve()
                 )
-        except (RuntimeError, ReferenceError, OSError) as e:
-            print(e)
+            except (RuntimeError, ReferenceError, OSError) as err:
+                print(err)
+    else:
+        bpy.ops.file.make_paths_absolute()
```
Comment (Tilix4) [inline openpype/hosts/blender/api/utils.py]: If always executed (no `break` in loop) it is recommended to keep it out of the `else`.

**Review by Tilix4 — CHANGES_REQUESTED** [review]
Nice! Some wonders

**Review by Tilix4 — APPROVED** [review]
LGTM! Thanks

**Comment by kaamaurice** [comment]
parse_known_args return a tuple ! code fixed.

**Pull request merged**